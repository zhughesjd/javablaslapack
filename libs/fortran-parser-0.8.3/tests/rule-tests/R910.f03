! Test read-stmt
!      read-stmt  is  READ (io-control-spec-list ) [ input-item-list ]
!                 or  READ format [, input-item-list ]
!
! Not tested here: io-control-spec-list, input-item-list, and format
read(10)
read(10, *) a, b

read 5, a, b, c

5 format (2e16.3, i5)

end

