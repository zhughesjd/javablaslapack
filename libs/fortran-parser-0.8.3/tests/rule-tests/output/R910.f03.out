*******************************************
args[1]: R910.f03
R1101:main-program__begin:
R 204:specification-part: numUseStmts=0 numImportStmts=0 numDeclConstructs=0
R 913:io-control-spec-list__begin:
R 406:int-literal-constant: digitString=[@12,219:220='10',<10>,6:5] kindParam=null
R 306:literal-constant:
R 701:primary:
R 913:io-control-spec-list: hasExpression=true keyword=null hasAsterisk=false
R 913:io-control-spec-list: count=1
R 910:read-stmt: hasInputItemList=false
R 214:action-stmt:
R 213:executable-construct:
R 913:io-control-spec-list__begin:
R 406:int-literal-constant: digitString=[@17,228:229='10',<10>,7:5] kindParam=null
R 306:literal-constant:
R 701:primary:
R 913:io-control-spec-list: hasExpression=true keyword=null hasAsterisk=false
R 913:io-control-spec-list: hasExpression=false keyword=null hasAsterisk=true
R 913:io-control-spec-list: count=2
R 915:input-item-list__begin:
R 613:part-ref: id=[@23,235:235='a',<209>,7:12] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R 603:designator: hasSubstringRange=false
R 601:variable:
R 915:input-item:
R 613:part-ref: id=[@26,238:238='b',<209>,7:15] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R 603:designator: hasSubstringRange=false
R 601:variable:
R 915:input-item:
R 915:input-item-list: count=2
R 910:read-stmt: hasInputItemList=true
R 214:action-stmt:
R 213:executable-construct:
R 209:execution-part-construct:
R 406:int-literal-constant: digitString=[@31,246:246='5',<10>,9:5] kindParam=null
R 306:literal-constant:
R 701:primary:
R 914:format:
R 915:input-item-list__begin:
R 613:part-ref: id=[@34,249:249='a',<209>,9:8] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R 603:designator: hasSubstringRange=false
R 601:variable:
R 915:input-item:
R 613:part-ref: id=[@37,252:252='b',<209>,9:11] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R 603:designator: hasSubstringRange=false
R 601:variable:
R 915:input-item:
R 613:part-ref: id=[@40,255:255='c',<209>,9:14] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R 603:designator: hasSubstringRange=false
R 601:variable:
R 915:input-item:
R 915:input-item-list: count=3
R 910:read-stmt: hasInputItemList=true
R 214:action-stmt:
R 213:executable-construct:
R 209:execution-part-construct:
R 313:label: lbl=[@43,258:258='5',<10>,11:0]
R1003:format-item-list__begin:
R1003:format-item: descOrDigit=[@-1,0:0='2e16.3',<194>,11:10] hasFormatItemList=false
R1003:format-item: descOrDigit=[@-1,0:0='i5',<194>,11:17] hasFormatItemList=false
R1003:format-item-list: count=2
R1002:format-specification: hasFormatItemList=true
R1001:format-stmt: label=[@43,258:258='5',<10>,11:0]
R 209:execution-part-construct:
R 208:execution-part:
R1103:end-program-stmt: id=null
R1101:main-program: hasProgramStmt=false hasExecutionPart=true hasInternalSubprogramPart=false
Parser exiting normally
********************************************
